core/commands/add.go

```diff
@@ -29,6 +31,7 @@ const progressReaderIncrement = 1024 * 256
 const (
 	progressOptionName = "progress"
 	wrapOptionName     = "wrap-with-directory"
+	hiddenOptionName   = "hidden"
```

Comment:
@chriscool 

> I am not very found of talking about "hidden" files. It think it is confusing to talk about both "ignored" and "hidden" files.

I agree that it is confusing. But it's also confusing for a Windows user to go `ipfs add . -r` and see a bunch of folders they didn't know existed. Just like Mac users are surprised to find the `.Ds_Store` files when they plug their USB drive into another computer.

So whose confusion do we want to suffer?

And if IPFS doesn't support one of the two notions, do we doom somebody else to have to write it instead?